labrad/node/__init__.py

```diff
@@ -503,95 +506,91 @@ class NodeServer(LabradServer):
     name = 'node %LABRADNODE%'
 
     def __init__(self, nodename, host, port, credential):
+        """Initialize a NodeServer instance.
+
+        Args:
+            nodename (str): The name of this node, e.g. from the LABRADNODE
+                environment variable.
+            host (str): The host where the labrad manager is running.
+            port (int): The port where the labrad manager is running.
+            credential (labrad.auth.Password): Credentials for connecting to the
+                labrad manager.
+        """
         LabradServer.__init__(self)
         self.nodename = nodename
         self.name = 'node %s' % nodename
         self.host = host
         self.port = port
         self.credential = credential
+
+        # Mapping from server name to ServerConfig that describes how to launch
+        # instances of the server.
         self.servers = {}
+
+        # Mapping from instance name to ServerProcess object. Instances are
+        # added to this mapping when they are successfully started and removed
+        # when they send a message indicating that the server process has
+        # stopped.
         self.instances = {}
-        self.initMessages(True)
+
+        # ServerProcess instances that will be notified when new servers connect
```

Comment:
It's not an issue with multiple servers, rather the `ServerProcess` object that launches server A as a subprocess wants to know when server A successfully connects to labrad; if server A doesn't connect before a timeout, we will kill the subprocess. So, the `ServerProcess` instance needs to be notified when servers connect so that it can see when its own server A connected, but we don't add server process instances to the `self.instances` dict until _after_ they are connected, so we need a different way to keep track of which `ServerProcess` instances need to be notified about new labrad servers.

I've got some changes in the pipeline that will rework all this to be much more sane, so it's not too important to understand fully how this stop-gap implementation works. We will refactor and test before cutting a new labrad release.